src/reflect/scala/reflect/internal/Types.scala

```diff
@@ -4752,7 +4752,7 @@ trait Types
   /** def isNonValueType(tp: Type) = !isValueElseNonValue(tp) */
 
   def isNonRefinementClassType(tpe: Type) = tpe match {
-    case SingleType(_, sym) => sym.isModuleClass
+    case SingleType(_, sym) => sym.isModuleOrModuleClass
```

Comment:
At the end, this seems to be suspiciously simple fix. No sure if that has other implications though.